features/device-config/service/src/main/java/org/opennms/features/deviceconfig/service/impl/DeviceConfigServiceImpl.java

```diff
@@ -69,6 +74,32 @@
     @Override
     public void triggerConfigBackup(String ipAddress, String location, String configType) throws IOException {
 
+        CompletableFuture<PollerResponse> future = pollDeviceConfig(ipAddress, location, configType);
+        future.whenComplete(((pollerResponse, throwable) -> {
+            if (throwable != null) {
+                LOG.info("Error while manually triggering config backup for IpAddress {} at location {} for config-type {}",
+                        ipAddress, location, configType, throwable);
+            }
+        }));
+    }
+
+    @Override
+    public CompletableFuture<byte[]> getDeviceConfig(String ipAddress, String location, String configType, int timeout) throws IOException {
+        CompletableFuture<byte[]> deviceConfigFuture = new CompletableFuture<>();
+        CompletableFuture<PollerResponse> future = pollDeviceConfig(ipAddress, location, configType);
+        try {
+            PollerResponse response = future.get(timeout, TimeUnit.MILLISECONDS);
+            if (response != null && response.getPollStatus() != null && response.getPollStatus().getDeviceConfig() != null) {
+                deviceConfigFuture.complete(response.getPollStatus().getDeviceConfig());
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            LOG.error("Error while getting device config for IpAddress {} at location {}", ipAddress, location, e);
+            deviceConfigFuture.completeExceptionally(e);
+        }
+        return deviceConfigFuture;
+    }
```

Comment:
This implementation has some issues:
* A future is returned, but in effect the future is already completed when it is returned. Returning the `byte[]` would be the same.
* The returned future is not completed in case the condition `response != null && response.getPollStatus() != null && response.getPollStatus().getDeviceConfig() != null` is not satisfied.

I propose to rewrite the method:

```suggestion
    public CompletableFuture<byte[]> getDeviceConfig(String ipAddress, String location, String configType, int timeout) throws IOException {
        return pollDeviceConfig(ipAddress, location, configType)
                .orTimeout(timeout, TimeUnit.MILLISECONDS)
                .thenApply(resp -> resp.getPollStatus().getDeviceConfig())
                .whenComplete((bytes, throwable) -> {
                    if (throwable != null) {
                        LOG.error("Error while getting device config for IpAddress {} at location {}", ipAddress, location, throwable);
                    }
                });
    }
```